src/controllers/portfolio/portfolio.ts

```diff
@@ -185,13 +187,64 @@ export class PortfolioController extends EventEmitter {
       ...t,
       flags: getFlags(res.data, 'gasTank', t.networkId, t.address)
     }))
+
+    let pinnedGasTankTokens: TokenResult[] = []
+
+    if (res.data.gasTank.availableGasTankAssets) {
+      const availableGasTankAssets = res.data.gasTank.availableGasTankAssets
+
+      pinnedGasTankTokens = availableGasTankAssets.reduce((acc: TokenResult[], token: any) => {
+        const isGasTankToken = !!gasTankTokens.find(
+          (gasTankToken: TokenResult) =>
```

Comment:
shouldn't we make this a closure, since we're using it a lot in this function

Reply:
What should we make a closure exactly and what are the pros of making the change?